src/index.js

```diff
@@ -59,7 +62,15 @@ const parsers = {
   typescript: {
     ...typescriptParsers.typescript,
     preprocess(text, opts) {
-      return organizeImports(text, path.extname(opts.filepath), path.dirname(opts.filepath))
+      let extname = '.ts'
+      let dirname = null
+
+      if (typeof opts.filepath === 'string') {
+        extname = path.extname(opts.filepath)
+        dirname = path.dirname(opts.filepath)
+      }
+
+      return organizeImports(text, extname, dirname)
```

Comment:
I find it funny that all of the patches I've had for this have all been for the typescript branch of the code - which BTW is all I use now too.

That said, I feel that the same logic should probably be on the babel config too if you don't mind adding it.

Reply:
Haha - lets see it as a positive point -> the typescript community uses and contributes to your library! 😄 

The JS part should already be safe, as `opts.filepath` will just evaluate to `undefined`. In that case the old `path.resolve(__dirname, '..', '..')` will kick in.

What do you think?

Reply:
Or do you mean instead of "fixing" the bug in here, to also extend the `babel` part to also automatically detect the extension / dir?

Reply:
Oh I think that it's safe, just that it likely has both this bug as well as the one that cause this behavior to change in 0.0.5.  But to be honest I'm happy to make things more consistent myself - you didn't cause this.

Reply:
let me run a few tests and I'll likely merge this and release a fix.